server/src/api/modules/players/player.service.js

```diff
@@ -113,8 +139,33 @@ async function getDataById(id) {
   }
 
   const latestSnapshot = await snapshotService.findLatest(player.id);
-
-  return { ...player.toJSON(), latestSnapshot: snapshotService.format(latestSnapshot) };
+  const {
+    attackExperience,
+    strengthExperience,
+    defenceExperience,
+    hitpointsExperience,
+    rangedExperience,
+    prayerExperience,
+    magicExperience
+  } = latestSnapshot;
+  const attackLevel = getLevel(attackExperience);
+  const strengthLevel = getLevel(strengthExperience);
+  const defenceLevel = getLevel(defenceExperience);
+  const hitpointsLevel = getLevel(hitpointsExperience);
+  const rangedLevel = getLevel(rangedExperience);
+  const prayerLevel = getLevel(prayerExperience);
+  const magicLevel = getLevel(magicExperience);
+  const combatLevel = getCombatLevel(
+    attackLevel,
+    strengthLevel,
+    defenceLevel,
+    hitpointsLevel,
+    rangedLevel,
+    prayerLevel,
+    magicLevel
+  );
```

Comment:
This should probably be put in a function rather than duplicating logic

Reply:
👍 Will look into making it less duplicating